Coalesce of health endpoint CIDRs

dezmodue opened the pull request:
Fixes: #18868. Multiple CIDRs are currently not coalesced for the health
endpoint when setting up routing the corresponding routing tables. This
results in orphaned routing entries that may conflict when IPs are
reused for workload pods after an agent restart.

Addresses comment https://github.com/cilium/cilium/pull/20112#issuecomment-1180343763

Signed-off-by: Simone Sciarrati <s.sciarrati@gmail.com>
Signed-off-by: Federico Hernandez <f@ederi.co>

gandro reviewed (approved):
Thank you for taking care of this!

recollir committed:
daemon: Coalesce endpoint CIDRs in ENI mode

Fixes: #18868. Multiple CIDRs are currently not coalesced for the health
endpoint when setting up routing the corresponding routing tables. This
results in orphaned routing entries that may conflict when IPs are
reused for workload pods after an agent restart.

Addresses comment https://github.com/cilium/cilium/pull/20112#issuecomment-1180343763

Signed-off-by: Simone Sciarrati <s.sciarrati@gmail.com>
Signed-off-by: Federico Hernandez <f@ederi.co>
